covid19_sfbayarea/data/hospitals.py

```diff
@@ -20,37 +21,119 @@
 CAGOV_BASEURL = "https://data.ca.gov"
 CAGOV_API = "/api/3/action/datastore_search"
 HOSPITALS_RESOURCE_ID = "42d33765-20fd-44b8-a978-b083b7542225"
-RESULTS_LIMIT = 50
+RESULTS_LIMIT = 200
 
 # For the output data
-SERIES_NAME = "Hospitalization"
+SERIES_NAME = "CA COVID-19 Hospitalization Data"
 BAYPD_META = "This data was pulled from the data.ca.gov CKAN Data API"
 
-logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+# =======================
+# Data Manipulation Utils
+# =======================
+
+
+def truncate_timestamp(timestamp: str) -> str:
+    """Truncate a timestamp to an ISO 8601-formatted date"""
+    truncated_timestamp = parse(timestamp).date().isoformat()
+    return truncated_timestamp
 
 
-def get_county(county: str) -> Dict:
-    """Return data just for the selected county. Include field notes.
-    This is basically an alias for `get_timeseries()` provided for consistency.
+def convert_null(record: Dict) -> Dict:
+    """Convert any null values to -1"""
+    for k, v in record.items():
+
+        if v is None:
+            record[k] = -1
+
+        else:
+            continue
+
+    return record
+
+
+def floats_to_ints(record: Dict) -> Dict:
+    """Convert zero-point floats for numeric fields to ints"""
+    fields: List = [
+        "all_hospital_beds",
+        "hospitalized_covid_confirmed_patients",
+        "hospitalized_covid_patients",
+        "hospitalized_suspected_covid_patients",
+        "icu_available_beds",
+        "icu_covid_confirmed_patients",
+        "icu_suspected_covid_patients",
+    ]
+
+    for field in fields:
+        val = record.get(field)
+
+        if val is None:
+            continue
+
+        else:
+            record[field] = int(val)
+
+    return record
+
+
+# ===================
+# Data Transformation
+# ===================
+
+
+def standardize_data(record: Dict) -> Dict:
+    """Transform certain data fields to make them conform to BAPD style
+
+    Specifically:
+    - truncate timestamps to ISO 8601-formatted dates
+    - remove "rank" field, if it exists
+    - convert all 'null' values to -1
+    - cast zero-point floats as int
+
+    Also, the key 'todays_date' is converted to 'report_date' for clarity.
     """
-    data = get_timeseries(county)
+    record["report_date"] = truncate_timestamp(record.pop("todays_date"))
```

Comment:
Two quick questions I had about this. Neither are really germane to the goal of wrapping this in a decent CLI, but I just happened to notice them when running the script this time. Neither are showstoppers:

1. Should we also be updating the field name in `meta_from_source`? Right now, the output looks like:

    ```js
    {
    "meta_from_source": [
        // ...
        {
        "info": {
            "notes": "Report date",
            "type_override": "timestamp",
            "label": "Today's Date"
        },
        "type": "timestamp",
        // Doesn't match the field name in the actual timeseries entries.
        "id": "todays_date"
        },
        // ...
    ],
    "series": {
        "san_francisco": [
        {
            "icu_covid_confirmed_patients": 24,
            "icu_suspected_covid_patients": 10,
            "hospitalized_covid_patients": -1,
            "hospitalized_suspected_covid_patients": 73,
            "icu_available_beds": 61,
            "county": "San Francisco",
            "hospitalized_covid_confirmed_patients": 50,
            "_id": 51,
            "all_hospital_beds": -1,
            "report_date": "2020-03-29"
        },
        // ...
    ```

2. Would this be better as just `date`, so it matches how the per-county data scraper formats time series?

Reply:
> Should we also be updating the field name in `meta_from_source`?

I'm a little torn on this, since it's represented as the metadata from the source (i.e. not us), but for clarity and consistency I think we probably should update to this:
``` js
    {
      "info": {
        "notes": "Report date",
        "type_override": "date",
        "label": "Date"
      },
      "type": "date",
      "id": "date"
    },
```
 
> Would this be better as just `date` [...] ?

Makes good sense. I'll make that change.

Reply:
@Mr0grog Question about the type hinting here. 

Here's the approach I've tried to updating the metadata:

```python
        # reflect renaming of the date field in the metadata
        updated_date_meta = {
            "info": {
                "notes": "Report date",
                "type_override": "date",
                "label": "Date"
            },
            "type": "date",
            "id": "date"
        }

        current_meta: List = ts_data.get("meta_from_source"))
        updated_meta: Dict = {"meta_from_source": []}

        for entry in current_meta:
            if entry.get("id") == "todays_date":
                index = ts_data["meta_from_source"].index(entry)
                updated_meta["meta_from_source"].insert(
                    index, updated_date_meta
                )
            else:
                updated_meta["meta_from_source"].append(entry)

        ts_data.update(updated_meta)
```

But with that, `mypy` complains about the `current_meta` variable assignment:
```
Incompatible types in assignment (expression has type "Optional[Any]", variable has type "List[Any]")
```

The simple fix seems to be:
```
current_meta = cast(List, ts_data.get("meta_from_source"))
```
... meaning I'm guaranteeing that will be a list. 

Is there a better way?

Reply:
That’s a good solution. If you are somewhere in the code where you can guarantee that a stubbed out `meta_from_source` is present, then I think indexed access (`current_meta = ts_data["meta_from_source"]`) may just work without any need for casting or explicitly specifying types, but it kind of depends on what types are defined where in your code.

(Side note: the way we are just manipulating `dicts` everywhere is really rough with typed code. It’s often better if you have classes ([data classes](https://docs.python.org/3/library/dataclasses.html) are especially nice) to represent the data model, because types can be better declared and enforced with them.)

Reply:
That worked - thanks! I think this is ready to merge now.